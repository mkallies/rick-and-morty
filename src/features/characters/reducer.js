import * as types from './constants'

const initialState = {
  characters: [],
  episodes: [],
  locations: [],
}

export default (state = initialState, action) => {
  switch (action.type) {
    case types.FETCH_CHARACTERS:
    case types.FETCH_EPISODES:
    case types.FETCH_LOCATION:
      return {
        ...state,
        [action.meta]: action.payload,
      }

    default:
      return state
  }
}
