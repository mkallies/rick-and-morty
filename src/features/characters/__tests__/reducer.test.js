import reducer from '../reducer'

describe('characters reducer', () => {
  test('should return the initial state', () => {
    const next = reducer(undefined, {})
    expect(next).toMatchSnapshot()
  })
})
