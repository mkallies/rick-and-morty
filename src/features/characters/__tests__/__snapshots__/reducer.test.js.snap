// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`characters reducer should return the initial state 1`] = `
Object {
  "characters": Array [],
  "episodes": Array [],
  "locations": Array [],
}
`;
